iothub_client/tests/iothubclient_fuzz_amqp/iothubclient_fuzz.c

```diff
@@ -0,0 +1,685 @@
+        {
+            (void)printf("ERROR: test file is to big\r\n");
+            return -1;
+        }
+
+        if (fuzzpacket_buffer[0] != (fuzzpacket_buffer[1] ^ 0xff))
+        {
+            // packe ID got fuzzed, ok to return success
+            return 0;
+        }
+
+        fuzzpacket_id = fuzzpacket_buffer[0];
+        (void)printf("fuzzing with packet id: %d\r\n", fuzzpacket_id);
+    }
+    else if (argc > 1)
+    {
+        (void)printf("Usage:\r\n");
+        return -1;
+    }
+
+
+    // Select the Protocol to use with the connection
+#ifdef SAMPLE_MQTT
+    protocol = MQTT_Protocol;
+#endif // SAMPLE_MQTT
+#ifdef SAMPLE_MQTT_OVER_WEBSOCKETS
+    protocol = MQTT_WebSocket_Protocol;
+#endif // SAMPLE_MQTT_OVER_WEBSOCKETS
+#ifdef SAMPLE_AMQP
+    protocol = AMQP_Protocol;
+#endif // SAMPLE_AMQP
+#ifdef SAMPLE_AMQP_OVER_WEBSOCKETS
+    protocol = AMQP_Protocol_over_WebSocketsTls;
+#endif // SAMPLE_AMQP_OVER_WEBSOCKETS
+#ifdef SAMPLE_HTTP
+    protocol = HTTP_Protocol;
+#endif // SAMPLE_HTTP
+
+    // Used to initialize IoTHub SDK subsystem
+    (void)IoTHub_Init();
+
+    IOTHUB_DEVICE_CLIENT_LL_HANDLE device_ll_handle;
+
+    (void)printf("Creating IoTHub Device handle\r\n");
+    // Create the iothub handle here
+    device_ll_handle = IoTHubDeviceClient_LL_CreateFromConnectionString(connectionString, protocol);
+    if (device_ll_handle == NULL)
+    {
+        (void)printf("Failure creating IotHub device. Hint: Check your connection string.\r\n");
+    }
+    else
+    {
+        // Set any option that are neccessary.
+        // For available options please see the iothub_sdk_options.md documentation
+
+#ifndef SAMPLE_HTTP
+        // Can not set this options in HTTP
+        bool traceOn = true;
+        IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_LOG_TRACE, &traceOn);
+#endif
+
+#if defined SAMPLE_MQTT || defined SAMPLE_MQTT_WS
+        //Setting the auto URL Encoder (recommended for MQTT). Please use this option unless
+        //you are URL Encoding inputs yourself.
+        //ONLY valid for use with MQTT
+        //bool urlEncodeOn = true;
+        //IoTHubDeviceClient_LL_SetOption(device_ll_handle, OPTION_AUTO_URL_ENCODE_DECODE, &urlEncodeOn);
+#endif
+
+        // Setting connection status callback to get indication of connection to iothub
+        (void)IoTHubDeviceClient_LL_SetConnectionStatusCallback(device_ll_handle, connection_status_callback, NULL);
+        (void)IoTHubDeviceClient_LL_SetMessageCallback(device_ll_handle, receive_msg_callback, NULL);
+        (void)IoTHubDeviceClient_LL_GetTwinAsync(device_ll_handle, deviceTwinCallback, NULL);
+        (void)IoTHubDeviceClient_LL_SetDeviceMethodCallback(device_ll_handle, device_method_callback, NULL);
+        (void)IoTHubDeviceClient_LL_SetDeviceTwinCallback(device_ll_handle, deviceTwinCallback, NULL);
+
+        int loop_count;
+        for (loop_count = 0; g_continueRunning && loop_count < 300000; loop_count++)
```

Comment:
`300000` arbitrary number? Just curious.

Reply:
yup, we do 30k message pumps to make sure all our work is done.